### PR TITLE
Ignore broken links and allow redirects

### PR DESCRIPTION
**Pull request by hugovk**
For #681.

* Group `linkcheck_*` configs together
* Update `linkcheck_ignore` rules
* Add `linkcheck_allowed_redirects` rules
* Update new PSF link redirect

Linkcheck docs: https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder

After this (and https://github.com/python/devguide/pull/1001 and https://github.com/python/devguide/pull/1002), all that fails with `make linkcheck` are the four tracked in https://github.com/python/devguide/issues/1003.

### CODE REVIEW

**Review by ezio-melotti — APPROVED** [review]

File: conf.py [hunk 1]
```diff
@@ -44,13 +44,55 @@
...
+    # The Discourse groups are private unless you are logged in
+    'https://discuss.python.org/groups/staff',
+    'https://discuss.python.org/groups/moderators',
+    'https://discuss.python.org/groups/admins',
+    # Anchor not found
+    r'https://github.com.+?#L\d+',
+    r'https://github.com/cli/cli#installation',
+    r'https://github.com/github/renaming#renaming-existing-branches',
+    # 403 Client Error: Forbidden
+    r'https://support.discord.com/hc/en-us/articles/219070107-Server-Nicknames',
```
Comment (CAM-Gerlach) [inline conf.py:93]: Can these not be fixed properly in the source?

**Review by CAM-Gerlach — COMMENTED** [review]
LGTM with one question

File: conf.py (hunk 1 above)
Comment (hugovk) [reply conf.py:93]: No, the links are working fine, for some reason, GitHub anchors cannot be found by the crawler:

```
(developer-workflow/c-api: line  173) broken    https://github.com/python/cpython/blob/2cd268a3a9340346dd86b66db2e9b428b3f878fc/Include/object.h#L592-L596 - Anchor 'L592-L596' not found
(developer-workflow/c-api: line  173) broken    https://github.com/python/cpython/blob/2cd268a3a9340346dd86b66db2e9b428b3f878fc/Objects/object.c#L2303-L2313 - Anchor 'L2303-L2313' not found
```
```
(getting-started/git-boot-camp: line  595) broken    https://github.com/cli/cli#installation - Anchor 'installation' not found
```
```
(getting-started/git-boot-camp: line  183) broken    https://github.com/github/renaming#renaming-existing-branches - Anchor 'renaming-existing-branches' not found
```

RTD has a similar rule for GitHub line anchors:

https://github.com/readthedocs/readthedocs.org/blob/c8f9b85eadb021adcd8f2196aba7dcea6dc820f1/docs/conf.py#L201


And Discord's website doesn't seem to allow robot crawlers:

```
(developer-workflow/communication-channels: line  217) broken    https://support.discord.com/hc/en-us/articles/219070107-Server-Nicknames - 403 Client Error: Forbidden for url: https://support.discord.com/hc/en-us/articles/219070107-Server-Nicknames
```

Comment (CAM-Gerlach) [reply conf.py:93]: Thanks, makes sense—can that be summarized in the comments above each, so that's clear to others in the future?

Comment (hugovk) [reply conf.py]: Sure! How's this?

Comment (CAM-Gerlach) [reply conf.py]: Looks pretty good, thanks :+1:

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @hugovk

**Pull request merged**